common/db.go

```diff
@@ -475,3 +531,113 @@ func buildSelectOrchsQuery(filter *DBOrchFilter) (string, error) {
 	}
 	return query, nil
 }
+
+// FindLatestMiniHeader eturns the MiniHeader with the highest blocknumber in the DB
+func (db *DB) FindLatestMiniHeader() (*blockwatch.MiniHeader, error) {
+	row := db.findLatestMiniHeader.QueryRow()
+	var (
+		number  int64
+		parent  string
+		hash    string
+		logsEnc string
+	)
+	if err := row.Scan(&number, &parent, &hash, &logsEnc); err != nil {
+		return nil, fmt.Errorf("could not retrieve latest header: %v", err)
+	}
+
+	logs, err := decodeLogsJSON([]byte(logsEnc))
+	if err != nil {
+		return nil, err
+	}
+	return &blockwatch.MiniHeader{
+		Number: big.NewInt(number),
+		Parent: ethcommon.HexToHash(parent),
+		Hash:   ethcommon.HexToHash(hash),
+		Logs:   logs,
+	}, nil
+}
+
+// FindAllMiniHeadersSortedByNumber returns all MiniHeaders in the DB sorting in descending order by block number
+func (db *DB) FindAllMiniHeadersSortedByNumber() ([]*blockwatch.MiniHeader, error) {
+	var headers []*blockwatch.MiniHeader
+	rows, err := db.findAllMiniHeadersSortedByNumber.Query()
+	if err != nil {
+		return nil, err
+	}
+	for rows.Next() {
+		var (
+			number  int64
+			parent  string
+			hash    string
+			logsEnc string
+		)
+		if err := rows.Scan(&number, &parent, &hash); err != nil {
+			return nil, err
+		}
+		logs, err := decodeLogsJSON([]byte(logsEnc))
+		if err != nil {
+			return nil, err
+		}
+		headers = append(headers, &blockwatch.MiniHeader{
+			Number: big.NewInt(number),
+			Parent: ethcommon.HexToHash(parent),
+			Hash:   ethcommon.HexToHash(hash),
+			Logs:   logs,
+		})
+	}
+	return headers, nil
+}
+
+// InsertMiniHeader inserts a MiniHeader into the database
+func (db *DB) InsertMiniHeader(header *blockwatch.MiniHeader) error {
+	if header == nil {
+		return errors.New("must provide a Miniheader")
+	}
+	if len(header.Hash) != 32 {
```

Comment:
I think we could be ok without the length check here since `common.Hash` is just a [type alias](https://github.com/ethereum/go-ethereum/blob/master/common/types.go#L47) for a `[]byte` of length 32.

Reply:
Fixed in ced063e